rmw_connextdds_common/src/common/rmw_impl_waitset_std.cpp

```diff
@@ -471,19 +496,22 @@ RMW_Connext_WaitSet::wait(
   rmw_events_t * const evs,
   const rmw_time_t * const wait_timeout)
 {
-  std::unique_lock<std::mutex> lock(this->mutex_internal);
-  if (this->waiting) {
-    RMW_CONNEXT_LOG_ERROR_SET(
-      "multiple concurrent wait()s not supported");
-    return RMW_RET_ERROR;
+  // std::unique_lock<std::mutex> lock(this->mutex_internal);
```

Comment:
The reason why `attach()` and `detach()` are not called from within the WS's critical section is that I have inverted the order of locking (Condition first, WS second), so it would introduce a deadlock risk with the code executed by the listeners.

Reply:
ah yes, thanks for explaining!